node/nodeTransactions.go

```diff
@@ -221,5 +205,36 @@ func (n *Node) prepareUnsignedTx(tx *smartContractResult.SmartContractResult) (*
 		Data:      string(tx.GetData()),
 		Code:      string(tx.GetCode()),
 		Signature: "",
+		Status:    status,
 	}, nil
 }
+
+func (n *Node) computeTransactionStatus(tx data.TransactionHandler, isInPool bool) string {
```

Comment:
We can return `TransactionStatus` here, and perhaps also use this type for the new tx field - and let the marshelizer do its job (optional - and hopefully it will work).

Reply:
json marshaler will be happy with that.

Reply:
yes maybe json marshalizer will be happy with that but if we change core.TransactionStatus in a more complex structure that contains more than a single `'string' `field will not work anymore. if you strongly suggest that ,i will change... wait for your answer.